### PR TITLE
Pass Problems By Props

### LINKED ISSUE #286: Pass Problem in by Props instead of Context
* This issue shouldn't be bad but requires lots of changes.. best to take it part by part
* It doesn't make sense that our problem is stored in the global context, as it only really needs to be used once on our coding page
* Instead use the `getProblem` API route to fetch the current problem from the database and make it a state object in `[id].js`.
    * You should be able to get the id through the router or something like that
    * Check out `Profile.jsx` to see how to pass the problem id into the API route and `_app.js` on how to handle the response properly
* Pass that object as a prop into all the components that need it, and also refactor those components to use that prop instead of the context

### PR DESCRIPTION
**Pull request by vwong031**
- Closes #286

### CODE REVIEW

File: pages/user/problems/[id].js [hunk 1]
```diff
@@ -58,6 +60,23 @@ const useResize = (containerRef, panelRef, initialWidth, minWidth = 0) => {
...
 const Problems = () => {
   const [state, setState] = useState(0);
+  const [problem, setProblem] = useState({});
+  const router = useRouter();
+  const { id } = router.query;
+
+  useEffect(() => {
+    const fetchProblem = async () => {
+      try {
+        axios
```
Comment (khandrew1) [inline pages/user/problems/[id].js]: add `await`

File: components/TestCases.jsx [hunk 2]
```diff
@@ -1,5 +1,4 @@
-import { useState, useContext } from "react";
-import CodeContext from "../components/CodeContext";
+import { useState } from "react";
 
 const TestCaseInput = ({ tests, state }) => {
   return (
```
Comment (khandrew1) [inline components/TestCases.jsx:4]: add a check here to make sure tests exist before rendering it, else just render an empty div; something along the lines of `{tests ? /* all that stuff */ : <div />}`

**Review by khandrew1 — CHANGES_REQUESTED** [review]

**Review by khandrew1 — APPROVED** [review]
